sources/Inputs/src/NetworkManager.cpp

```diff
@@ -140,6 +144,40 @@ NetworkManager::buildTree() {
       LOG(debug) << "Node " << bus1->getID() << " connected to " << bus2->getID() << " and " << bus3->getID() << " by 3W " << (*it_t)->getID() << LOG_ENDL;
     }
   }
+
+  const auto& hvdcLines = network->getHvdcLines();
+  for (const auto& hvdcLine : hvdcLines) {
+    const auto& converter_dyn_1 = hvdcLine->getConverter1();
+    const auto& converter_dyn_2 = hvdcLine->getConverter2();
+    if (!converter_dyn_1->getInitialConnected() || !converter_dyn_2->getInitialConnected()) {
+      continue;
+    }
+    auto converter_1 = ConverterInterface(converter_dyn_1->getID(), converter_dyn_1->getBusInterface()->getID());
+    auto converter_2 = ConverterInterface(converter_dyn_2->getID(), converter_dyn_2->getBusInterface()->getID());
+
+    HvdcLine::ConverterType converterType;
+    if (converter_dyn_1->getConverterType() == DYN::ConverterInterface::ConverterType_t::VSC_CONVERTER) {
+      converterType = HvdcLine::ConverterType::VSC;
+      auto vsc_converter_dyn_1 = boost::static_pointer_cast<DYN::VscConverterInterface>(converter_dyn_1);
```

Comment:
Since we are using pointers, it is better to use dynamic_pointer_cast here to check the inheritance

Reply:
I am not sure why. If we are sure of the underlying type is it not better to use the static_pointer_cast ?

Reply:
static cast will always work since we are dealing with pointers, while dynamic cast will not

Reply:
dynamimc cast is safer as it makes a type check but less efficient performance wise. However we call this once so maybe using dynamic cast is better indeed

Reply:
once by hdvc line

Reply:
yes, let me rephrase: I meant that this algorithm is called once at the beginning of the flow and Hvdc lines are not numerous so it should be OK

Reply:
But we check it beforehand with the if condition so we are sure this cast will suceed, that's exactly when we want to use a static pointer cast

Reply:
OK

Reply:
The check before is independant from the type of the object

Reply:
How is it independent ? The if condition checks if the type of the object is VSC_CONVERTER

Reply:
The type attribute and the type of the object are independant. You can imagine a bug where the wrong type attribute is given to a structure type

Reply:
If we don't trust the code that has already been tested a bit, then I can not imagine the number of tests we would put every 10 lines of code.

Reply:
OK